Index: mdm-2.20.10/daemon/getvt.c
===================================================================
--- mdm-2.20.10.orig/daemon/getvt.c	2010-02-27 17:27:46.956530497 +0100
+++ mdm-2.20.10/daemon/getvt.c	2010-02-27 17:27:58.617031143 +0100
@@ -149,7 +149,7 @@ mdm_get_vt_device (int vtno)
 #ifdef __sun
 #define MDMCONSOLEDEVICE "/dev/vt/0"
 #else
-#define MDMCONSOLEDEVICE "/dev/console"
+#define MDMCONSOLEDEVICE "/dev/tty0"
 #endif
 
 
